Expensify/App
# Run migrations and refresh app state if relevant

CC: @luacmartins (since you worked on this recently) @roryabraham @marcaaron 
### HOLD
- [x] Needs [this PR](https://github.com/Expensify/Web-Expensify/pull/33122) to be on production, or else the new API command will error out.

### Details

We will run FixAccount like we do in OldDot and reinitialize data if something changed for the user or we ran migrations. If FixAccount returns no changes (which it will 90% of the time), we will do nothing. This will work without refreshing the whole app state and users won't need to log out and log back in.

### Fixed Issues
$ https://github.com/Expensify/Expensify/issues/195477

### Tests
Same as QA but done locally (with the other Web-E PR linked above active)


### QA Steps
You can reach out to `amal@expensify.com` @TomatoToaster to test this on staging if needed.

1. Find an account that was made before 03/28/2022 (your expensifail should work)
2. If you haven't logged into OldDot since the above date on this account that is ideal.
3. Go to staging.new.expensify.com and open up the dev console and make sure you have verbose enabled:
<img width="703" alt="image" src="https://user-images.githubusercontent.com/19364431/166998692-caab703b-40dc-475a-8f60-77a30f176763.png">

4. Filter the console commands by `FixAccount`
<img width="755" alt="image" src="https://user-images.githubusercontent.com/19364431/167003977-80d9534b-ac93-4724-9448-4ee0b12ee1e5.png">


5. Now log into your expensifail account. Verify that you see these commands fire off (If you don't see the third one, that's fine for this step, that means you likely logged into this account in OldDot some time after 03/28/2022):
<img width="755" alt="image" src="https://user-images.githubusercontent.com/19364431/167003904-de265e83-03e6-49f6-8a5e-5526d1a86252.png">


6. In another tab, log into the same account on OldDot, and then run this snippet in the console:
`NVP.set('expensify_migration_2022_03_28_policyExpenseChatMigration')` Then refresh the page on NewDot and verify that the three commands in the above screenshot are there.

7. Refresh the page one more time and verify you only see the first 2 commands:
<img width="756" alt="image" src="https://user-images.githubusercontent.com/19364431/167002759-e189781d-c511-4f76-ade1-1619efcc0734.png">




### Tested On
